docs/standard/serialization/binaryformatter-migration-guide/winforms-migration-guide.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "BinaryFormatter Migration Guide: WinForms Migration Guide"
+description: "Learn about the effects of the deprecation and removal of BinaryFormatter from .NET on Windows Forms and how to migrate."
+ms.date: 7/31/2024
+no-loc: [BinaryFormatter, WinForms]
+helpviewer_keywords:
+  - "BinaryFormatter"
+  - "WinForms"
+---
+
+# Window Forms migration guide for BinaryFormatter
+
+## BinaryFormatter removal
+
+Starting in .NET 9, `BinaryFormatter` has moved to a separate, unsupported NuGet package due to its known [security risks](../binaryformatter-security-guide.md). For more information about the risks `BinaryFormatter` poses and the reason for its removal, see [BinaryFormatter migration guide](overview.md). With BinaryFormatter’s removal, it's expected that many Windows Forms applications will be impacted, and you'll need to take action to complete your migration to .NET 9 or a later version.
+
+## How BinaryFormatter affects Windows Forms
+
+Windows Forms itself has and still contains `BinaryFormatter` code internally for scenarios such as clipboard, drag-and-drop, and storing or loading resources at design time. However, measures have been taken to mitigate the usage of `BinaryFormatter` behind the scenes for common primitive types, including primitive collections. Usage of these types in clipboard and drag-and-drop scenarios will continue to work without using `BinaryFormatter` and without any gesture from the developer. Primitive types include the following:
```

Comment:
>Primitive types include the following

`Primitive types` term is interpreted differently depending on context. I.e. these are not C# primitive types - https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/built-in-types. 

These are NRBF `primitive types` - https://learn.microsoft.com/en-us/dotnet/api/system.formats.nrbf.primitivetyperecord?view=net-8.0 

Consider either giving more context or calling them common types.